### PR TITLE
MM-27826: Test revoking own and others sessions.

### PR DESCRIPTION
**Pull request by mkraft**
### Summary

E2E tests for revoking own and others sessions via the system console.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-27826

### CODE REVIEW

**Review by hahmadia — APPROVED** [review]
Code looks good! Just one question is that those two functions you've added, are they able to be used outside of the file? Does it make sense to make them global functions rather than just functions within the file? Just a thought

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 1]
```diff
@@ -147,6 +149,67 @@ describe('User Management', () => {
         });
     });
 
+    it('MM-T938 Users - Revoke a user\'s session', () => {
+        // # Create an active session for the user we're going to revoke.
+        apiLogin(testUser.username, testUser.password);
```
Comment (furqanmlk) [inline e2e/cypress/integration/system_console/user_management_spec.js:154]: this function is already exist
```cy.apiLogin({username: 'sysadmin', password: 'secret'});```

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 2]
```diff
@@ -147,6 +149,67 @@ describe('User Management', () => {
...
+            cy.wrap(response.data).should('be.empty');
+        });
+    }
+
+    function revokeSessionByEmailAndRole(email, role) {
+        // # Go to the user management page in the system console.
+        cy.visit('/admin_console/user_management/users').wait(TIMEOUTS.ONE_SEC);
+
+        // # Search for a test user.
+        cy.get('#searchUsers').clear().type(email).wait(TIMEOUTS.HALF_SEC);
```
Comment (furqanmlk) [inline e2e/cypress/integration/system_console/user_management_spec.js:196]: Test-Library best practise recommendations,
```cy.findByPlaceholderText('Search users'); ```

**Review by furqanmlk — CHANGES_REQUESTED** [review]
@mkraft 
these two test case are failing as user does not exist,
Please see the screenshot, am I missing something ?

image

File: e2e/cypress/integration/system_console/user_management_spec.js (hunk 1 above)
Comment (mkraft) [reply e2e/cypress/integration/system_console/user_management_spec.js]: > @mkraft
> these two test case are failing as user does not exist,
> Please see the screenshot, am I missing something ?
> 
> image

Can you try again with this latest update? It turns out some of the other tests update the email addresses that these new tests were looking for.

File: e2e/cypress/integration/system_console/user_management_spec.js [hunk 3]
```diff
@@ -147,9 +149,70 @@ describe('User Management', () => {
...
+        // # Login to the UI as system_admin.
+        cy.apiAdminLogin();
+
+        revokeSessionByUsernameAndRole(testUser.username, 'Member');
+
+        // # Ensure the user has no session.
+        checkNoSessionsForUser(testUser.id);
+    });
+
+    it('MM-T938 Users - Revoke own session', () => {
```
Comment (furqanmlk) [inline e2e/cypress/integration/system_console/user_management_spec.js:165]: @mkraft 
This test case id is `MM-T939`

**Review by furqanmlk — CHANGES_REQUESTED** [review]
@mkraft 
I am wondering how did you create multiple sessions of a user,
Test case step#03 - Find the other user you've logged in with multiple sessions
or am I missing something here. 
Please advise

**Pull request closed without merging**